packages/common/src/audius-query/audius-query.test.ts

```diff
@@ -0,0 +1,87 @@
+import { beforeEach, describe, it, expect, vi } from 'vitest'
+
+import { ID } from '~/models'
+
+import { createApi } from './createApi'
+
+const mockContext = {
+  dispatch: vi.fn()
+}
+
+describe('audius-query', function () {
+  describe('request batcher', function () {
+    const fetchBatchSpy = vi.fn(async ({ ids }: { ids: ID[] }, _: any) => {
+      return ids.map((id) => ({ track_id: id }))
+    })
+
+    const testApi = createApi({
+      reducerPath: 'testApi',
+      endpoints: {
+        getById: {
+          fetch: async ({ id, myArg }: { id: ID; myArg?: string }, _: any) => {
+            return { id }
+          },
+          fetchBatch: fetchBatchSpy,
+          options: {
+            idArgKey: 'id',
+            schemaKey: 'track'
+          }
+        }
+      }
+    })
+
+    beforeEach(() => {
+      fetchBatchSpy.mockClear()
+    })
+
+    it('batches requests when fetchBatch is specified', async function () {
+      const item1 = testApi.fetch.getById({ id: 1, myArg: 'test' }, mockContext)
+      const item2 = testApi.fetch.getById({ id: 2, myArg: 'test' }, mockContext)
+
+      expect(await item1).toEqual({ track_id: 1 })
+      expect(await item2).toEqual({ track_id: 2 })
+      expect(fetchBatchSpy).toHaveBeenCalledTimes(1)
+      expect(fetchBatchSpy).toHaveBeenCalledWith(
+        { ids: [1, 2], myArg: 'test' },
+        mockContext
+      )
+    })
+
+    it('performs multiple batches when requests fall outside of the batch period', async function () {
```

Comment:
how about test for when a new batch queues up during the fetch of the current one? might be harder to orchestrate though.

Reply:
I like that idea, let me see what I can do